File: eunji/0x05/BOJ_1874.py
N = int(input())
ans = []
stack = []

count = 1
flag = 1
for i in range(N):
    x = int(input())
    while (count <= x):
        stack.append(count)
        count += 1
        ans.append('+')
        
    if (stack[-1] == x):
        stack.pop()
        ans.append('-')
    else:
        flag = 0
        
if flag == 0: print("NO")
else:
    for i in range(len(ans)):
        print(ans[i])

File: eunji/0x05_stack/BOJ_1874.cpp
#include <iostream>
#include <stack>
#include <vector>
using namespace std;

int main(void){
    ios::sync_with_stdio(0);
    cin.tie(0);
    int N;
    cin >> N;
    stack<int> S;
    vector<char> ans;
    int cnt = 1;
    for (int i=0; i<N; i++){
        int x;
        cin >> x;

        while (cnt <= x){
            S.push(cnt);
            cnt ++;
            ans.push_back('+');
        }
        if (S.top() == x){
            S.pop();
            ans.push_back('-');
        }
        else{
            cout << "NO";
            return 0;
        }
    }


    for (int i = 0; i < ans.size(); i++) {
        cout << ans[i] << "\n";
    }
}